crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -865,6 +872,38 @@ impl<'interner> TypeChecker<'interner> {
         }
     }
 
+    fn bind_function_type_impl(
+        &mut self,
+        fn_params: &Vec<Type>,
+        fn_ret: &Type,
+        callsite_args: &Vec<(Type, ExprId, Span)>,
+        span: Span,
+        skip_params: usize,
+    ) -> Type {
+        let real_fn_params_count = fn_params.len() - skip_params;
```

Comment:
We should remove the `skip_params` as well, since we don't need the environment parameter in the function's parameter types. It can be added later during monomorphization when we need it there but there's not much use storing it in two places in the function type.

Reply:
fixed